### PR TITLE
WRP-26628: Kitten browser: Update State and Data management sections

### PR DESCRIPTION
**Pull request by vJIYEv**
Enact-DCO-1.0-Signed-off-by: Jiye Kim (jiye.kim@lge.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Update old-styled code in Kitten Browser

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
The old-styled code in Kitten Browser is updated.
Other code in Kitten Browser is also updated not only 'state and data management' section.

**App Setup**
- Add explanation of directory structure
  - There was not any explanation of directory structure before, so it could make readers confused.

**Repeaters and Lists**
- Change the comment of the code for explaining destructuring
  - It was difficult to understand the meaning of the comment.
- Change the position of 'note'
  - Reading the 'note' after the explain of destructuring is better to understand the content of the 'note'.

**State and Data Management**
- Change the code in this section for better understanding
  - index -> indexPanel, kitten -> indexKitten
  - onSelectKitten -> onChangeKitten, onSelectKitten (case by case)
  - onSelect -> onSelectKitten, onSelect (case by case)
- Add and change the descriptions
  - Add and change description about onSelect, onSelectKitten, onChangeKitten  
  - Add description about 'Changeable' for better understanding

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRP-26628

### Comments
Enact-DCO-1.0-Signed-off-by: Jiye Kim (jiye.kim@lge.com)

### CODE REVIEW

**Comment by vJIYEv** [comment]
travis cli install branch is changed to master branch
```
install:
    - git clone --branch=master --depth 1 https://github.com/enactjs/cli ../cli
``` 
should return to develop branch

File: src/pages/docs/tutorials/tutorial-kitten-browser/app-setup/index.md [hunk 1]
```diff
@@ -119,7 +117,7 @@ export {
 You might be wondering why the `<img />` tag uses the self-closing syntax (`/>`). This is a requirement imposed by JSX, so whenever you include an element that doesn't have any children, you must either use the self-closing syntax or explicitly close it (`<img src=""></img>`).
 
 Comments in your code are helpful and JSX is no different. Comments in JSX must be within an expression to be correctly parsed by the plugin.
-If you'd like to know more information, see [jsx-in-depth](https://reactjs.org/docs/jsx-in-depth.html).
+If you'd like to know more information, see [here](https://react.dev/learn/javascript-in-jsx-with-curly-braces).
```
Comment (0x64) [inline src/pages/docs/tutorials/tutorial-kitten-browser/app-setup/index.md:120]: It looks better to put the title of the document rather than 'here'.

File: src/pages/docs/tutorials/tutorial-kitten-browser/app-setup/index.md [hunk 2]
```diff
@@ -143,7 +141,7 @@ Compiled with warnings.
 Search for the keywords to learn more about each warning.
 To ignore, add // eslint-disable-next-line to the line before.
 ```
-The `npm run serve` command (aliased as `enact serve`) will build and host your project on a web server. While building the code, it allows ESLint to perform a static analysis of the source code. ESLint helps you to find syntax errors, anti-patterns, and to write the source code in a consistent code style. See [about ESLint](https://eslint.org/docs/about/) for more information. If you only want to run a static analysis of the code you wrote, use `npm run lint` command (aliased as `enact lint`).
+The `npm run serve` command (aliased as `enact serve`) will build and host your project on a web server. While building the code, it allows ESLint to perform a static analysis of the source code. ESLint helps you to find syntax errors, anti-patterns, and to write the source code in a consistent code style. See [about ESLint](https://eslint.org/docs/latest/) for more information. If you only want to run a static analysis of the code you wrote, use `npm run lint` command (aliased as `enact lint`).
```
Comment (0x64) [inline src/pages/docs/tutorials/tutorial-kitten-browser/app-setup/index.md:144]:
```suggestion
The `npm run serve` command (aliased as `enact serve`) will build and host your project on a web server. While building the code, it allows ESLint to perform a static analysis of the source code. ESLint helps you to find syntax errors, anti-patterns, and to write the source code in a consistent code style. Visit [ESLint](https://eslint.org/) for more information. If you only want to run a static analysis of the code you wrote, use `npm run lint` command (aliased as `enact lint`).
```

File: src/pages/docs/tutorials/tutorial-kitten-browser/data-and-state/index.md [hunk 3]
```diff
@@ -14,58 +14,58 @@ order: 5
...
+In the [previous step](../panels/), we focused on the app structure with Panels. Next, we'll investigate how custom and native events are used to notify your app of user actions, how state is defined for the app, and how to use events to update the state of your app. We'll wrap up with a brief discussion on how to manage state data.
 
 ## Events in React
 
-[Native events in React](https://reactjs.org/docs/events.html) are registered using the camelCase version (`onMouseDown`) of their native name (`onmousedown`). If you wish to be notified of an event, you can pass a function as the value for the appropriate prop. The function will receive a synthetic event as its first argument, which is a cross-browser wrapper around the original event.
+[Native events in React](https://react.dev/reference/react-dom/components/common) are registered using the camelCase version (`onMouseDown`) of their native name (`onmousedown`). If you wish to be notified of an event, you can pass a function as the value for the appropriate prop. The function will receive a synthetic event as its first argument, which is a cross-browser wrapper around the original event.
```
Comment (0x64) [inline src/pages/docs/tutorials/tutorial-kitten-browser/data-and-state/index.md:21]:
```suggestion
[Native events in React](https://react.dev/learn/responding-to-events) are registered using the camelCase version (`onMouseDown`) of their native name (`onmousedown`). If you wish to be notified of an event, you can pass a function as the value for the appropriate prop. The function will receive a synthetic event as its first argument, which is a cross-browser wrapper around the original event.
```

File: src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md [hunk 4]
```diff
@@ -126,11 +126,11 @@ propTypes: {
 	size: PropTypes.number
 },
 ```
-We've introduced a couple new ES6 features in this update. The unique function parameter in the computed property above is an example of destructuring and the backtick string is a template literal. Both are covered in more detail below.
+We've introduced a couple JavaScript features in this update. The unique function parameter in the computed property above is an example of destructuring and the backtick string is a template literal. Both are covered in more detail below.
 
 ### Destructuring
 
-[Destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) ([spec](https://262.ecma-international.org/6.0/#sec-destructuring-assignment)) is a new language feature of ES6 that allows you to extract the properties of an object into local variables. The basic syntax is:
+[Destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) ([spec](https://262.ecma-international.org/6.0/#sec-destructuring-assignment)) is a language feature of JavaScript that allows you to extract the properties of an object into local variables. The basic syntax is:
```
Comment (0x64) [inline src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md:133]:
```suggestion
[Destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) ([spec](https://262.ecma-international.org/6.0/#sec-destructuring-assignment)) allows you to extract the properties of an object into local variables. The basic syntax is:
```

File: src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md [hunk 5]
```diff
@@ -141,21 +141,25 @@ const {a: myA, b} = obj; // == const myA = obj.a, b = obj.b;
 ```
 As you saw above, you can also destructure an object received as an argument to a function:
 ```js
-url: ({index, size}) => {	// == url: (props) => { const index = props.index, size = props.size;
+url: ({index, size}) => {
+	// == const {index, size} = props;
+	// == const index = props.index, size = props.size;
```
Comment (0x64) [inline src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md:146]: Let's revert this change. The comment could mislead readers if the comment is not on the same lime as destructuring.

File: src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md [hunk 6]
```diff
@@ -141,21 +141,25 @@ const {a: myA, b} = obj; // == const myA = obj.a, b = obj.b;
...
 	return `//loremflickr.com/${size}/${size}/kitten?random=${index}`;
 }
 ```
+`index` prop is the index of kitten from [`Repeater` in `List` component](./#repeater). The function argument is a object which contains props of `Kitten` component, so `index` and `size` props in the function argument is destructured to the local variables named `index` and `size`.
+
```
Comment (0x64) [inline src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md:151]: I want to know what this phrase explains.

File: src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md [hunk 7]
```diff
@@ -141,21 +141,25 @@ const {a: myA, b} = obj; // == const myA = obj.a, b = obj.b;
...
 Destructuring is particularly useful in React because your render methods and computed properties will receive their input as a property object, which you can deconstruct into the relevant properties.
 
 ### Template Literals
 
-[Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) ([spec](https://262.ecma-international.org/6.0/#sec-template-literals)) are a new string type in ES6 that support multi-line strings, expression interpolation, and tagged templates.
+[Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) ([spec](https://262.ecma-international.org/6.0/#sec-template-literals)) are a string type in JavaScript that support multi-line strings, expression interpolation, and tagged templates.
```
Comment (0x64) [inline src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md:156]: Actually, `template literals` are NOT strings. Commonly called `template strings` due to its common usage but it's literal. Let's find more exact description.

File: src/pages/docs/tutorials/tutorial-kitten-browser/app-setup/index.md (hunk 2 above)
Comment (0x64) [reply src/pages/docs/tutorials/tutorial-kitten-browser/app-setup/index.md:144]: The link should be an intro page as our intention is to guide readers to find out what ESLint is.

File: src/pages/docs/tutorials/tutorial-kitten-browser/data-and-state/index.md (hunk 3 above)
Comment (0x64) [reply src/pages/docs/tutorials/tutorial-kitten-browser/data-and-state/index.md:21]: In the context of following phrase, the link is to guide how to use events in React not to list up possible events.

File: src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md (hunk 4 above)
Comment (0x64) [reply src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md:133]: Maybe it looks natural?

File: src/pages/docs/tutorials/tutorial-kitten-browser/app-setup/index.md (hunk 1 above)
Comment (vJIYEv) [reply src/pages/docs/tutorials/tutorial-kitten-browser/app-setup/index.md]: I changed 'here' to 'JavaScript in JSX with Curly Braces'.

File: src/pages/docs/tutorials/tutorial-kitten-browser/app-setup/index.md (hunk 2 above)
Comment (vJIYEv) [reply src/pages/docs/tutorials/tutorial-kitten-browser/app-setup/index.md]: I accepted your suggestion.

File: src/pages/docs/tutorials/tutorial-kitten-browser/data-and-state/index.md (hunk 3 above)
Comment (vJIYEv) [reply src/pages/docs/tutorials/tutorial-kitten-browser/data-and-state/index.md]: I accepted your suggestion.

File: src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md (hunk 4 above)
Comment (vJIYEv) [reply src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md]: I accepted your suggestion.

File: src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md (hunk 5 above)
Comment (vJIYEv) [reply src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md]: I accepted your suggestion.

File: src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md (hunk 6 above)
Comment (vJIYEv) [reply src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md]: I reverted this change.

File: src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md (hunk 7 above)
Comment (vJIYEv) [reply src/pages/docs/tutorials/tutorial-kitten-browser/lists/index.md]: I changed 'a string type' to 'literals' by referring to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals).

File: src/pages/docs/tutorials/tutorial-kitten-browser/data-and-state/index.md [hunk 8]
```diff
@@ -14,58 +14,58 @@ order: 5
...
 
 React doesn't provide an explicit implementation of custom events. In Enact, custom events follow React's event name pattern of having an `on` prefix followed by the name of the event. Like native events, custom events will receive the event payload as the first argument but it will be a simple object, not a synthetic event.
 
 ## Defining our State
 
-For our App, we have two pieces of state we need to manage: the active panel and the selected kitten. Both can be represented as numbers representing the index of the panel and the index of the kitten in `kittens`. We will also define two custom events, `onNavigate` and `onSelectKitten`, that we can use to indicate when each of those indices should change.
+For our App, we have two pieces of state we need to manage: the active panel (`List` or `Detail`) and the selected kitten. Both can be represented as numbers representing the index of the panel and the index of the kitten in `kittens`. We will also define two custom events, `onPanelIndexChange` and `onKittenIndexChange`, that we can use to indicate when each of those indices should change.
```
Comment (seunghoh) [inline src/pages/docs/tutorials/tutorial-kitten-browser/data-and-state/index.md:27]: Don't like to use (),
the active panel (`List` or `Detail`) and the selected kitten
-> the active panel, which consists of List and Detail, and the selected kitten.
or just simply
-> the active panel and the selected kitten

File: src/pages/docs/tutorials/tutorial-kitten-browser/index.md [hunk 9]
```diff
@@ -6,7 +6,7 @@ order: 4
 ## Introduction
 Before you begin this tutorial, make sure you have created a new Enact project.  See [Enact Development Setup](../setup/) for more information.  You can delete `src/views/MainPanel.js` as it will not be used in the tutorial example.
 
-Kitten Browser is a tutorial that demonstrates many of the features of the Enact framework, ES6 and React that will be useful to you.  This is more in-depth than [Hello, Enact!](../tutorial-hello-enact/).
+Kitten Browser is a tutorial that demonstrates many of the features of the Enact framework, JavaScript and React that will be useful to you.  This is more in-depth than [Hello, Enact!](../tutorial-hello-enact/).
```
Comment (seunghoh) [inline src/pages/docs/tutorials/tutorial-kitten-browser/index.md:9]:
```suggestion
Kitten Browser is a tutorial that demonstrates many of the features of the Enact framework and React that will be useful to you.  This is more in-depth than [Hello, Enact!](../tutorial-hello-enact/).
```

Comment (vJIYEv) [reply src/pages/docs/tutorials/tutorial-kitten-browser/index.md]: I accepted your suggestion.

File: src/pages/docs/tutorials/tutorial-kitten-browser/data-and-state/index.md (hunk 8 above)
Comment (vJIYEv) [reply src/pages/docs/tutorials/tutorial-kitten-browser/data-and-state/index.md]: I accepted your suggestion.
(the active panel, which consists of List and Detail, and the selected kitten.)

**Comment by vJIYEv** [comment]
Code in samples is updated by https://github.com/enactjs/samples/pull/207 to match with this PR
